### PR TITLE
Add a CONTRIBUTING.md file

### LINKED ISSUE #93: Add CONTRIBUTING.md file
Should have info for people hacking on bindgen (whereas README.md is more for users of bindgen / landing page).

As a newcomer, I'd really like to have an architectural overview of the project.

### PR DESCRIPTION
**Pull request by fitzgen**
It provides an overview of how to file a helpful issue and how to run tests.

Fixes #93 

r? @emilio

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+## Code of Conduct <span id="coc"/>
+
+We abide by the [Rust Code of Conduct][coc] and ask that you do as well.
+
+[coc]: https://www.rust-lang.org/en-US/conduct.html
+
+## Filing an Issue <span id="issue"/>
+
+Think you've found a bug? File an issue! To help us understand and reproduce the
+issue, provide us with:
```
Comment (emilio) [inline CONTRIBUTING.md]: `try to provide us`? Some issues like #67 are really tricky to minimize :)

Worth commenting here maybe that running with RUST_LOG=bindgen spits a log really helpful for debugging.

Comment (fitzgen) [inline CONTRIBUTING.md]: Oo good thought on the logging

**Pull request merged**